pkg/signature/options.go

```diff
@@ -55,3 +56,50 @@ type VerifyOption interface {
 	RPCOption
 	MessageOption
 }
+
+type signerVerifierOpts struct {
```

Comment:
A quick comment but will add more later - can we perhaps put these under appropriate `SignOption` and `VerifierOption` above, and then have a `SignerVerifierOption` interface that combines both?

Reply:
also, I'd rather keep this file as a list of interfaces, and move any specific implementations into the `options` subdirectory

Reply:
> A quick comment but will add more later - can we perhaps put these under appropriate SignOption and VerifierOption above, and then have a SignerVerifierOption interface that combines both?

How would that work?

I'd imagine we have these functions in the interface:
```golang
	ApplyHash(*crypto.Hash)
	ApplyRSAPSSOptions(*rsa.PSSOptions)
	ApplyIsED25519ph(*bool)
```

however, where should I put them? They are not strictly used while signing and they are not strictly used while verifying, but they are used when the SignerVerifier objects are created.

> also, I'd rather keep this file as a list of interfaces, and move any specific implementations into the options subdirectory

Sure!

Reply:
OK, I took a bit more time and reviewed the bulk of the code again. I think what was confusing to me initially (which you stated, but I overlooked) that these new options are relevant at initialization time only, whereas all the other options are used at the time of actual signing/verification operations.

Given that realization, I feel a bit more strongly about defining these in:
* a separate `.go` file (i.e. not in options.go)
* having a different interface name (`LoadOptions`)?

Reply:
I also noticed there is some overlap with `CryptoSignerOpts` too, that you can specify a hash there. I agree that splitting this out helps to distinguish between at-init and at-use options. We might want to note that there are multiple ways of specifying a hash too.

Reply:
Ok perfect, I will move them in separate go file! Is `options` dir still a good place for it?

@haydentherapper are you suggesting to implement `CryptoSignerOpts` as well or were you just noticing that there is an overlap?

Reply:
Just noticing overlap, that a hash alg can be overridden per signing event.